### PR TITLE
Test fixed routing comparator

### PR DESCRIPTION
**Pull request by shiliyu**

### CODE REVIEW

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import org.apache.http.util.EntityUtils;
+import org.junit.jupiter.api.BeforeEach;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class AbstractHandlerIntegrationTest {
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java:17]: Can you make `PreHandlerIntegrationTest` extends it and remove the duplicate code?

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Inject
+    @Config("forest.http.port")
+    Integer port;
+
+    @BeforeEach
+    void setUp() {
+        client = HttpClients.createDefault();
+        router.traces.clear();
+    }
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java:30]:
```
@AfterEach 
void cleanUp() {
    client.close();
}
```

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    public CloseableHttpResponse sendHttpRequest(String method, String path) {
+        String uri = "http://localhost:" + port + path;
+        HttpUriRequest request = RequestBuilder.create(method).setUri(uri).build();
+        CloseableHttpResponse response;
+        try {
+            response = client.execute(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java]: This is test, you can throw everything out and don't catch anything. Not a big issue, though.

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        return result;
+    }
+
+    public String assert200(Supplier<CloseableHttpResponse> excutable) {
+        return assertStatuscode(excutable, 200);
+    }
+
+    public String assert404(Supplier<CloseableHttpResponse> excutable) {
+        return assertStatuscode(excutable, 404);
+    }
```
Comment (blindpirate) [inline core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java]: I would prefer:

```
class HttpResponse {
   int statusCode;
   String body;
   void assert200() {
   }
   void assert404() {
   }
   // Or even 
   // void assertStatusCode(int expectedStatusCode) {}
}

...

public HttpResponse sendRequest(String url) {
        CloseableHttpResponse response = excutable.get();
        return new HttpResponse(statusCode, EntityUtils.toString(response.getEntity());
    }
```

**Review by blindpirate — CHANGES_REQUESTED** [review]

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java (hunk 1 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java:17]: Definitely!

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java (hunk 3 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java]: alright.

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java (hunk 4 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java]: yes, this makes more sense.

File: core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java (hunk 2 above)
Comment (shiliyu) [reply core/src/test/java/io/forestframework/core/http/routing/AbstractHandlerIntegrationTest.java:30]: will fix it.

**Review by blindpirate — APPROVED** [review]

**Pull request merged**